File: samples/android/src/main/java/com/example/jsonapi/resource/Article.java
package com.example.jsonapi.resource;

import jsonapi.Id;
import jsonapi.Resource;
import jsonapi.ToMany;
import jsonapi.ToOne;

import java.util.List;

@Resource(type = "articles")
public class Article {

  @Id final String id;
  final String title;
  @ToOne(name = "author") final Person author;
  @ToMany(name = "comments") final List<Comment> comments;

  public Article(String id, String title, Person author, List<Comment> comments) {
    this.id = id;
    this.title = title;
    this.author = author;
    this.comments = comments;
  }

  public String headline() {
    return this.title + " by " + this.author.getName();
  }

  @Override
  public String toString() {
    return "Article{" +
      "id='" + id + '\'' +
      ", title='" + title + '\'' +
      ", author=" + author +
      ", comments=" + comments +
      '}';
  }
}

File: samples/jvm/src/main/java/com/example/jsonapi/resource/Article.kt
package com.example.jsonapi.resource

import jsonapi.Id
import jsonapi.Resource
import jsonapi.ToMany
import jsonapi.ToOne

@Resource("articles")
data class Article(
  @Id val id: String?,
  var title: String? = "",
  @ToOne("author") val author: Person? = null,
  @ToMany("comments") val comments: List<Comment>? = null,
)
